metabase/metabase
# Pass uuids as strings to SQL Server

Closes https://github.com/metabase/metabase/issues/51189

### Description

Azure Synapse doesn't like it when you give it a uuid parameter, even though synapse does support uuids in general. Converting those uuids to strings beforehand fixes this error, and regular SQL Server is also fine with that.

For background, this issue was introduced when we bumped mssql-jdbc from version 12.6.1 to 12.8.1 [here](https://github.com/metabase/metabase/commit/9fe5a85e910081f645bef39b8b4090b038009caa) . The [changelog for 12.7](https://github.com/microsoft/mssql-jdbc/blob/main/CHANGELOG.md) mentions "Added support for TDSType.GUID", which is presumably what caused the issue.

### How to verify

If you create some sample data involving uuids in a sql server db, you can at least verify that this change doesn't break regular sql server.  However, verifying that the original bug is fixed requires signing up for azure synapse, which is somewhat more work.

### Checklist
